server/game/basecard.ts

```diff
@@ -609,7 +609,7 @@ class BaseCard extends EffectSource {
         if(!parent || parent.getType() !== CardTypes.Character || !ignoreType && this.getType() !== CardTypes.Attachment) {
             return false;
         }
-        if(this.anyEffect(EffectNames.AttachmentMyControlOnly) && context.player !== parent.controller) {
+        if(this.anyEffect(EffectNames.AttachmentMyControlOnly) && context.player !== parent.controller && this.controller !== parent.controller) {
```

Comment:
I don't like that we compare `context.player` in one test here and `this.controller` in the other.  They should both be the same...

Reply:
Any suggestions on how to use the same one for both?  Using Rimei, `context.player` is player1, `parent.controller` is player2, and `this.controller` is player2.

It looks like steal effects (ex: Karada) are relying on the `context.player !== parent.controller` bit.  Not sure why.